File: Src/ECScape.Core/Engine/UiInterface.cs
namespace ECScape.Core.Engine;

public static class UiInterface
{
    private static readonly int InterfaceHeight = 3;
    private static int Width;
    private static int Height;

    public static void SetBounds(int width, int height)
    {
        Width = width;
        Height = height;
    }

    public static int InterfaceStart = 0;
    public static int InterfaceEnd = InterfaceHeight;

    public static int TotalWidth => Width;
    public static int TotalHeight => Height;

    public static int WorldWidth => TotalWidth;
    public static int WorldHeight => TotalHeight - 3;

    public static int WorldTop => InterfaceHeight;
    public static int WorldBottom => Height - 1;
}

File: Web/src/core/uiInterface.js
const UiInterface = {
    InterfaceHeight: 30,
    Width: 0,
    Height: 0,

    setBounds: function (width, height) {
        this.Width = width;
        this.Height = height;
    },

    get InterfaceStart() { return 0; },
    get InterfaceEnd() { return this.InterfaceHeight; },

    get TotalWidth() { return this.Width; },
    get TotalHeight() { return this.Height; },

    get WorldWidth() { return this.TotalWidth; },
    get WorldHeight() { return this.TotalHeight - 30; },
    get WorldTop() { return this.InterfaceHeight; },
    get WorldBottom() { return this.Height - 1; }
};

export default UiInterface;